doc/overlays.xml

```diff
@@ -62,37 +59,40 @@ the overlay, and add a symbolic link to it in the
 <section xml:id="sec-overlays-layout">
 <title>Overlays Layout</title>
 
-<para>An overlay is a Nix expression, which is a function which accepts 2
-arguments.</para>
+<para>Overlays are expressed as Nix functions which accept 2 arguments and return a set of
+packages</para>
 
 <programlisting>
 self: super:
 
 {
-  foo = super.foo.override { ... };
-  bar = import ./pkgs/bar {
-    inherit (self) stdenv fetchurl;
-    inherit (self) ninja crawl dwarf-fortress;
+  boost = super.boost.override {
+    python = self.python3;
+  };
+  rr = super.callPackage ./pkgs/rr {
```

Comment:
E.g. I would think we need to provide a custom `callPackage` for each `self` layer using
`lib.callPackagesWith`.